library/core/src/iter/adapters/flatten.rs

```diff
@@ -310,6 +310,7 @@ where
 /// Real logic of both `Flatten` and `FlatMap` which simply delegate to
 /// this type.
 #[derive(Clone, Debug)]
+#[unstable(feature = "trusted_len", issue = "37572")]
```

Comment:
Unexpected to see it there

Reply:
From what I see `FlattenCompat` is in a `where` clause of a public item, so it's considered leaked (reachable).
Requirements for stability attributes are pretty conservative, they are required on reachable items even if they are not nameable (exported).